backend/routes/meta.js

```diff
@@ -273,14 +376,15 @@
 			});
 	}
 
-	function getAllIdsWith4DSequenceTag(req, res, next) {
+	function getMeshIdsByRules(req, res, next) {
+		const rules = req.body;
 		let branch;
 
 		if (!req.params.rev) {
 			branch = C.MASTER_BRANCH_NAME;
 		}
 
-		ModelHelpers.getAllIdsWith4DSequenceTag(req.params.account, req.params.model, branch, req.params.rev)
+		Meta.getMeshIdsByRules(req.params.account, req.params.model, branch, req.params.rev, rules)
 			.then(obj => {
 				responseCodes.respond(utils.APIInfo(req), req, res, next, responseCodes.OK, obj, undefined, req.param.rev ? config.cachePolicy : undefined);
```

Comment:
can't define cache policy here you'd end up caching previous results which could've been a completely different rule.